kv/txn_correctness_test.go

```diff
@@ -777,3 +777,32 @@ func TestTxnDBWriteSkewAnomaly(t *testing.T) {
 	checkConcurrency("write skew", onlySerializable, []string{txn1, txn2}, verify, true, t)
 	checkConcurrency("write skew", onlySnapshot, []string{txn1, txn2}, verify, false, t)
 }
+
```

Comment:
Instead of adding a test to txn_correctness_test.go, it should be added to txn_test.go.

txn_correctness_test.go is very expensive and is meant to exhaustively check concurrency anomalies. This is really just a bug in the implementation you found (good job, btw!). With a normal unittest, you can exactly script out the situation you've highlighted below.